### PR TITLE
Add AnalysisServicesTestRunner and replace AsTestsBase

### PR DESCRIPTION
**Pull request by CaptainFanZzz**
## Description

Add AnalysisServicesTestRunner and replace AzTestsBase

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/AnalysisServices/AnalysisServices.Test/ScenarioTests/AnalysisServicesTestRunner.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                .WithCommonPsScripts(new[]
+                {
+                    @"Common.ps1",
+                    @"../AzureRM.Resources.ps1",
+                })
+                .WithNewRmModules(helper => new[]
+               {
+                    helper.RMProfileModule,
+                    helper.GetRMModulePath("Az.AnalysisServices.psd1"),
+                    helper.GetRMModulePath("AzureRM.AnalysisServices.psd1")
```
Comment (ziyuezh576) [inline src/AnalysisServices/AnalysisServices.Test/ScenarioTests/AnalysisServicesTestRunner.cs:39]: same as the line above, delete it.

File: src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ public class DataPlaneCommandTests : AsTestsBase
                                         + "65kxhZWVUbTHaPuEvg03ZQ3esDb6wxQewJPAL-GARg6S9wIN776Esw8-53AWhzFu0fIut-9FXGma6jV7"
                                         + "MYPoUUcFuQzLZgphecPyMPXSVhummVCdBwX9sizxnmFA";
 
-        public DataPlaneCommandTests(ITestOutputHelper output)
+        public DataPlaneCommandTests(ITestOutputHelper output) : base(output)
```
Comment (ziyuezh576) [inline src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs:64]: File name contains Data Plane, and the file doesn't look like a TestController, so I don't think the file needs to be edited.

File: src/AnalysisServices/AnalysisServices.sln [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27703.2042
+# Visual Studio Version 17
+VisualStudioVersion = 17.1.32414.318
```
Comment (ziyuezh576) [inline src/AnalysisServices/AnalysisServices.sln:3]: don't edit solution file

File: src/AnalysisServices/AnalysisServices.Test/ScenarioTests/AsTests.cs [hunk 4]
```diff
@@ -18,91 +18,88 @@ namespace Microsoft.Azure.Commands.AnalysisServices.Test.ScenarioTests
...
+    public class AsTests : AnalysisServicesTestRunner
     {
-        public XunitTracingInterceptor _logger;
-
-        public AsTests(Xunit.Abstractions.ITestOutputHelper output)
+        public AsTests(Xunit.Abstractions.ITestOutputHelper output) : base(output)
         {
-            _logger = new XunitTracingInterceptor(output);
-            XunitTracingInterceptor.AddToContext(_logger);
+
```
Comment (ziyuezh576) [inline src/AnalysisServices/AnalysisServices.Test/ScenarioTests/AsTests.cs:25]: delete the empty line

**Review by ziyuezh576 — CHANGES_REQUESTED** [review]

**Review by ziyuezh576 — CHANGES_REQUESTED** [review]

File: src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs (hunk 2 above)
Comment (ziyuezh576) [reply src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs:64]: Sorry I didn't notice that AsTestsBase class had been removed.

File: src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs [hunk 5]
```diff
@@ -61,7 +61,7 @@ public class DataPlaneCommandTests : AsTestsBase
...
         {
             XunitTracingInterceptor.AddToContext(new XunitTracingInterceptor(output));
```
Comment (ziyuezh576) [inline src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs:66]: Remove this line. It's removed in other test classes, and doing so will not fail any tests.

**Review by ziyuezh576 — CHANGES_REQUESTED** [review]

**Review by ziyuezh576 — APPROVED** [review]

File: src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs [hunk 6]
```diff
@@ -36,7 +36,7 @@
 
 namespace Microsoft.Azure.Commands.AnalysisServices.Test.InMemoryTests
 {
-    public class DataPlaneCommandTests : AsTestsBase
+    public class DataPlaneCommandTests : AnalysisServicesTestRunner
```
Comment (vidai-msft) [inline src/AnalysisServices/AnalysisServices.Test/InMemoryTests/DataPlaneCommandTests.cs:39]: Please make it inherit from RMTestBase instead of TestRunner

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Review by vidai-msft — APPROVED** [review]

**Pull request merged**